### DISCUSSION TITLE
[Suggestion] Flux::toast()->flash()

### DISCUSSION THREAD

**Discussion by donnyxray** [livewire/flux#139] [Feature Requests]
I'm replacing Filament Notifications with Flux::toast. Unfortunately, unlike the Filament Notifications, the Flux toasts do not survive redirects. It would be great if we can flash toast's through the session.

Sample use case:
* User hits /downloads
* I redirect them to /premium-downloads because they're a paying customer
* User should see a toast noting they were redirected

**Comment by sewid** [comment]
The suggested solution is not working for me when using full page components / volt. I try to set a toast notification before a redirect. The notification is visible for a part of a second before the redirect occurs.

The persisted toast notification is part of my app.blade.php - the layout file for the full page components.

**Comment by donnyxray** [comment]
Does not work for me either. Perhaps it works when the toast and redirect are done through Livewire, but my issue is needing to send the toast when hitting a non-Livewire URL that redirects. Hence the need to use a session flash.

**Comment by sewid** [comment]
> The suggested solution is not working for me when using full page components / volt. I try to set a toast notification before a redirect. The notification is visible for a part of a second before the redirect occurs.
> 
> The persisted toast notification is part of my app.blade.php - the layout file for the full page components.

I was wrong, the solution is working. I made an error with the redirect.

I used: `$this->redirectRoute('my_route', true);`, correct is: `$this->redirectRoute('my_route', [], true, true);`

Maybe someone has the same error...

**Reply by Uhasith** [reply]
This worked for me ;)

**Comment by joshhanley** [comment]
Thanks for reporting! We have converted this issue to a feature request for now, so we can focus on fixing any bugs first.

**Comment by RiadhKHEDHIRI** [comment]
Please help. I can't make the persistent toast work.

I am using a fresh Laravel installation with Breeze Livewire functional API. I added

```
...
  @persist('toast')
    <flux:toast />
  @endpersist
  @fluxScripts
</body>
</html>
```

to `resources/layouts/app.blade.php`.

I added `Flux::toast('Your changes have been saved.', duration: 0);` to the `$updateProfileInformation` function in `update-profile-information-form.blade.php`, and when I click on the `Dashboard` link, which is `wire:navigate,` the toast disappears.

Any hint?

**Reply by RiadhKHEDHIRI** [reply]
I use Firefox on Windows

**Comment by gustavosobrinho01** [comment]
I have the same issue; I’ve tried everything and haven’t been successful.

**Reply by jeffchown** [reply]
Same here.  I cannot get my toasts to persist across a redirect.

**Comment by RiadhKHEDHIRI** [comment]
When inspecting the new page, the toast is here with my custom content. For some reason, it is not shown. Maybe some of the show conditions are no longer met. I am thinking of values erased or not persisted.

**Comment by RiadhKHEDHIRI** [comment]
This is persisted

```
        @persist('toaster')
        <script>
            function incrementValue()
            {
                var value = parseInt(document.getElementById('number').value, 10);
                value = isNaN(value) ? 0 : value;
                value++;
                document.getElementById('number').value = value;
            }
        </script>
        <form>
            <input type="text" id="number" value="0"/>
            <input type="button" onclick="incrementValue()" value="Increment Value" />
        </form>
        @endpersist
```

**Comment by calebporzio** [comment]
This has been fixed in Livewire. You can now persist toast elements between `wire:navigate` navigations and redirects using `@persist`.

https://github.com/livewire/livewire/releases/tag/v3.5.10

**Reply by RiadhKHEDHIRI** [reply]
Thank you!

**Reply by RiadhKHEDHIRI** [reply]
It works!
